src/snowflake/connector/url_util.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All rights reserved.
+#
+
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import re
+import urllib.parse
+from logging import getLogger
+
+logger = getLogger(__name__)
+
+
+class SnowflakeURLUtil:
```

Comment:
This doesn't need to be class, just define `url_validator` at global lever (Python standard says to make it all caps to signify that it's a constant too). So that'd be something like
```
URL_VALIDATOR = re.compile(
    "^http(s?)\\:\\/\\/[0-9a-zA-Z]([-.\\w]*[0-9a-zA-Z@:])*(:(0-9)*)*(\\/?)([a-zA-Z0-9\\-\\.\\?\\,\\&\\(\\)\\/\\\\\\+&%\\$#_=@]*)?$"
)

def is_valid_url(url: str) -> bool:
    return bool(URL_VALIDATOR.match(url))
```